core/src/main/java/com/yahoo/imapnio/async/data/SearchResult.java

```diff
@@ -6,20 +6,34 @@
 import javax.annotation.Nullable;
 
 /**
- * This class provides the list of message sequence numbers from search command response.
+ * This class provides the modification sequence and the list of message sequence numbers from search command response.
  */
 public class SearchResult {
     /** Search command response sequence number, could be message sequence or UID. */
-    @Nonnull
     private final List<Long> msgNumbers;
 
+    /** Modification sequence, is only shown when CondStore is enabled. */
+    private final Long highestModSeq;
+
     /**
      * Initializes a {@code SearchResult} object with message number collection.
      *
      * @param msgNumbers collection of message number from search command result
      */
     public SearchResult(@Nonnull final List<Long> msgNumbers) {
```

Comment:
You may want to create a constructor with no args to a handle (msgNumbers == null) case

Reply:
where do you use this constructor? am I missing something? I do not see getting used.

This is a response I think.